backend/src/main/kotlin/outcobra/server/web/auth/util/JwtUtil.kt

```diff
@@ -0,0 +1,62 @@
+package outcobra.server.web.auth.util
+
+import io.jsonwebtoken.Claims
+import io.jsonwebtoken.JwtException
+import io.jsonwebtoken.Jwts
+import io.jsonwebtoken.SignatureAlgorithm
+import me.mkweb.releasr.web.auth.exception.JwtExpiredException
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Component
+import outcobra.server.util.getExpirationTime
+import outcobra.server.util.setExpirationTime
+import outcobra.server.web.auth.model.OutcobraUser
+import java.time.LocalDateTime
+
+
+@Component
+class JwtUtil(@Value("\${security.jwt.secret}") private val secret: String) {
+    /**
+     * Tries to parse specified String as a JWT token. If successful, returns User object with username, id and role prefilled (extracted from token).
+     * If unsuccessful (token is invalid or not containing all required user properties), simply returns null.
```

Comment:
* same as below (id and role won't be prefilled)
* the jwt not containing required user properties is not checked and should therefore also be removed from the documentation